### PR TITLE
Provide the ability to Mock WorkflowActivityContext

### PR DESCRIPTION
**Pull request by siri-varma**
# Description

Made the WorkflowActivityContext abstract.
Created DaprWorkflowActivityContext  and made it inherit from WorkflowActivityContext 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1200

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

**Comment by WhitWaldo** [comment]
@siri-varma It looks like the DCO check failed. 

Could you please make sure each of your commits are signed?

**Comment by WhitWaldo** [comment]
@siri-varma Looks like we've got several build errors regarding nullability. If you can fix those, I'd appreciate it!

**Comment by siri-varma** [comment]
@WhitWaldo let me know if anything else is needed on this PR. Thank you

**Review by WhitWaldo — CHANGES_REQUESTED** [review]
Just a couple of nits and we're good here.

File: src/Dapr.Workflow/WorkflowActivityContext.cs [hunk 1]
```diff
@@ -13,29 +13,21 @@
...
-
         /// <summary>
         /// Gets the name of the activity.
         /// </summary>
-        public TaskName Name => this.innerContext.Name;
+        public abstract TaskName Name { get; }
 
         /// <summary>
-        /// Gets the unique ID of the current workflow instance.
+        /// Gets the unique ID of thcurrent workflow instance.
```
[the remark this answers is not in the document]
Comment (siri-varma) [reply src/Dapr.Workflow/WorkflowActivityContext.cs]: @WhitWaldo done resolved

**Review by WhitWaldo — APPROVED** [review]

**Comment by WhitWaldo** [comment]
Thank you very much for your contribution @siri-varma !

**Comment by marcduiker** [comment]
@holopin-bot @siri-varma Thanks so much this contribution! 🚀

**Pull request merged**